src/Claim.test.ts

```diff
@@ -1,335 +1,424 @@
-      author: TheRaven.attributes.author,
-      name: TheRaven.attributes.name,
-    })
-
-    const claimId1 = await getClaimId(work1).catch(returnError)
-    const claimId2 = await getClaimId(work2).catch(returnError)
+    const expectedMessage = 'Cannot sign a claim whose id has been altered or generated incorrectly.'
 
     assert({
-      given: 'two claims with disordered keys',
-      should: 'have the same claims id',
-      actual: claimId1 === claimId2,
-      expected: true,
+      given: 'a claim with an altered id',
+      should: `throw an error with the message ${expectedMessage}`,
+      actual: await sign({ ...TheRaven, id: 'be81cc75bcf6ca0f1fdd356f460e6ec920ba36ec78bd9e70c4d04a19f8943102' }).catch(
+        getErrorMessage
+      ),
+      expected: expectedMessage,
     })
   }
 
   {
-    const work1: Claim = makeClaim({
-      name: TheRaven.attributes.name,
-      author: TheRaven.attributes.author,
-    })
-
-    const work2: Claim = makeClaim({
-      Author: TheRaven.attributes.author,
-      NAME: TheRaven.attributes.name,
+    const expectedMessage = 'Cannot sign a claim with an invalid creator in the signing options.'
+    const invalidSign = signClaim({
+      algorithm: SigningAlgorithm.Ed25519Signature2018,
+      privateKeyBase58: testPrivateKeyEd25519Base58,
     })
 
-    const claimId1 = await getClaimId(work1).catch(returnError)
-    const claimId2 = await getClaimId(work2).catch(returnError)
-
     assert({
-      given: 'two claims with keys casing',
-      should: 'NOT have the same claims id',
-      actual: claimId1 !== claimId2,
-      expected: true,
+      given: 'a claim with creator undefined',
+      should: `throw an error with the message ${expectedMessage}`,
+      actual: await invalidSign(TheRaven).catch(getErrorMessage),
+      expected: expectedMessage,
     })
   }
 
   {
-    const signature = await getClaimSignature(TheRaven, PrivateKeyEAP)
-
-    assert({
-      given: 'a signature of a Claim',
-      should: 'be the signature equal to of work signature',
-      actual: signature,
-      expected: TheRaven.signature,
+    const invalidSign2 = signClaim({
+      algorithm: SigningAlgorithm.Ed25519Signature2018,
+      creator: 'data:,someoneelse',
+      privateKeyBase58: testPrivateKeyEd25519Base58,
     })
-  }
 
-  {
-    const expectedMessage = 'Cannot sign a claim that has an empty .id field.'
+    const signedClaim = await invalidSign2({
+      ...TheRaven,
+      '@context': { ...DefaultClaimContext, ...DefaultWorkClaimContext },
+    }).catch(e => e)
 
     assert({
-      given: 'a claim without id',
-      should: `throw an error with the message ${expectedMessage}`,
-      actual: await getClaimSignature({ ...TheRaven, id: '' }, PrivateKeyEAP).catch(returnError),
-      expected: new Error(expectedMessage),
+      given: 'an invalid public key in creator',
+      should: `create a claim that can not be verified`,
+      actual: await isValidSignature(signedClaim),
```

Comment:
Can signClaim throw an Error instead of allowing the signing with an invalid public key?

Reply:
Or maybe like you mentioned before, we generate the public key instead of taking it, either of those solutions is a better user experience IMO.

Reply:
We don't want to generate the public key in this instance. If they provide an issuer with a public key that doesn't match the signature, then the signature should be invalid.

I would have to deep dive into the jsonld-signatures library to change this behavior - OR validate the signature as part of the signing process. The latter probably makes more sense here.

Reply:
Yup, no arguments against isValidSignature not providing the public keys. This test is a signClaim test though, the signClaim allows for the user to provide a privateKey and publicKey that do not match.

Reply:
I can see generating a public key if none is provided; however, if a provided public key does not match the signature, I don't think we should generate a correct one for them.

Reply:
Agreed ( that is not what I meant above if that's the impression I gave  )